### PR TITLE
Maielo/upgrade socket io

### PR DESCRIPTION
**Pull request by maielo**
- upgrade of collaboration server of socket io to latest version
- slight refactor
- better descriptions

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "@types/debug": "4.1.5",
     "@types/express": "4.17.11",
     "@types/node": "14.14.31",
-    "@types/socket.io": "2.1.4",
+    "@types/socket.io": "^3.0.2",
```
Comment (dwelle) [inline package.json:8]: the package should have its own types now so let's remove

File: .env.development [hunk 2]
```diff
@@ -1 +1,2 @@
-PORT=3002
\ No newline at end of file
+PORT=3002
+CORS_ORIGIN=http://localhost:3000
```
Comment (dwelle) [inline .env.development:2]: I'd remove in dev, since people can run the local on different ports (I run excal on 5000)

File: src/index.ts [hunk 3]
```diff
@@ -28,67 +28,86 @@ server.listen(port, () => {
...
+    // handlePreflightRequest: (req, res) => {
+    //   const headers = {
+    //     "Access-Control-Allow-Headers": "Content-Type, Authorization",
+    //     "Access-Control-Allow-Origin":
+    //       (req.header && req.header.origin) || "https://excalidraw.com",
+    //     "Access-Control-Allow-Credentials": true,
+    //   };
+    //   res.writeHead(200, headers);
+    //   res.end();
+    // },
```
Comment (dwelle) [inline src/index.ts:49]: remove

File: .env.development (hunk 2 above)
Comment (maielo) [reply .env.development:2]: i will elave there empty values

File: package.json (hunk 1 above)
Comment (maielo) [reply package.json:8]: removed

**Pull request merged**